pkg/targets/adapter/ibmmqtarget/adapter.go

```diff
@@ -86,13 +102,22 @@ func (a *ibmmqtargetAdapter) Start(ctx context.Context) error {
 }
 
 func (a *ibmmqtargetAdapter) dispatch(ctx context.Context, event cloudevents.Event) (*cloudevents.Event, cloudevents.Result) {
+	ceTypeTag := metrics.TagEventType(event.Type())
+	ceSrcTag := metrics.TagEventSource(event.Source())
+
+	start := time.Now()
+	defer func() {
+		a.sr.ReportProcessingLatency(time.Since(start), ceTypeTag, ceSrcTag)
+	}()
+
 	var msg []byte
 
 	if a.mqEnvs.DiscardCEContext {
 		msg = event.Data()
 	} else {
 		jsonEvent, err := json.Marshal(event)
 		if err != nil {
+			a.sr.ReportProcessingError(false, ceTypeTag, ceSrcTag)
```

Comment:
Shouldn't `userManaged` (first function parameter) be `true` if we're using the replier?

cc. @odacremolbap

Reply:
I'm not completely sure what `userManaged` is stands for, so if you say so, I can make it `true`

Reply:
It means that the replier will respond with "200" and an error payload. This bypasses Knative's delivery mechanism, and therefore the error is considered to be "user-managed".

But in this case, I would like a confirmation from @odacremolbap that the replier is actually doing what I described.

Reply:
@antoineco is right.

If you want the event to be re-delivered by Kn you can use a different replier method (that no one has used so far), but if you don't want the sender to re-send the event, and in this case I don't think you want that because the same payload will fail again when marshaling, you use the `a.replier.Error` method that you are using.

That method returns 200, meaning ACK, message received so that it is not re-delivered again.
But it also returns a payload to sender containing an error structure and an attribute that metrics can use to mark as `user-managed` error.

Reply:
@odacremolbap wasn't this behavior configurable through the env variable? I see the point in preventing retries on error but in some cases, I'd like to be able to utilize Knative's dead letter sink option (if I cannot change sender logic, for example).

Reply:
If you use Knative's features you get the full pack retries + DLS.
If you want to rely on Knative's you can use replier's [ErrorKnativeManaged](https://github.com/triggermesh/triggermesh/blob/d80e46844aff67ab9c927b1cddfcf59848127092/pkg/targets/adapter/cloudevents/replier.go#L114) method.

Replier has no configuration via environment.